### ISSUE TITLE
DWC Import - eventDate not letting me import without leading zeros

### ISSUE THREAD

**Issue by tmcelrath** [SpeciesFileGroup/taxonworks#1763]
**Describe the bug**

When I import a DWCA and it includes eventDate without a leading zero, the import fails

**To Reproduce**
1. When I create a DWCA import
2. Then I try to import a record with an event date without a leading zero (e.g. 2019-5-19 instead of 2019-05-19 or 2019-12-1 instead of 2019-12-01)
3. Record will not import

Error:
![image](https://user-images.githubusercontent.com/11411328/93910780-2fde2300-fcc7-11ea-89ee-c2fda1bec1c7.png)

Field in question:
![image](https://user-images.githubusercontent.com/11411328/93910825-3c627b80-fcc7-11ea-88e7-5b3d7cd4bffa.png)

**Expected behavior**
Lack of leading zeros should not affect import.

**Environment (please identify where you experience this bug:**
*** One of ***
- Sandworm - McElrath Test Project
- Chrome

**Comment by LocoDelAssembly** [comment]
This happens because I use this function to parse eventDate:
```ruby
2.7.1 :002 > Date.iso8601('2019-5-19')
Traceback (most recent call last):
        5: from /home/hernan/.rvm/rubies/ruby-2.7.1/bin/irb:23:in `<main>'
        4: from /home/hernan/.rvm/rubies/ruby-2.7.1/bin/irb:23:in `load'
        3: from /home/hernan/.rvm/rubies/ruby-2.7.1/lib/ruby/gems/2.7.0/gems/irb-1.2.3/exe/irb:11:in `<top (required)>'
        2: from (irb):2
        1: from (irb):2:in `iso8601'
Date::Error (invalid date)
2.7.1 :003 > Date.iso8601('2019-05-19')
 => #<Date: 2019-05-19 ((2458623j,0s,0n),+0s,2299161j)> 
```

Doesn't have options to relax formatting so might need to use something completely different.

**Comment by mjy** [comment]
In some ways I want to relax, in others I don't (it's less sloppy to enforce it overall).  At minimum we'd want to point users to a Excel macro that pads date columns, or show them how to do it.

**Comment by tmcelrath** [comment]
At the very least, if this is why it's failing, we need a message saying that they need to add leading zeros.

**Comment by mjy** [comment]
@tmcelrath Does the DwC validator mention anything?

**Comment by tmcelrath** [comment]
Just tested against it - no it does not.

**Comment by tmcelrath** [comment]
Okay this is especially frustrating b/c day month year fields do not allow leading zeros. So basically I need leading zeros in one place and can't have them in another.

**Comment by mjy** [comment]
What is the other?

**Comment by tmcelrath** [comment]
"month" and "day" cannot have leading zeros
![image](https://user-images.githubusercontent.com/11411328/93932193-6a56b880-fce5-11ea-9e6d-f8dbabffb28b.png)

**Comment by mjy** [comment]
@LocoDelAssembly 'verbatimEventDate' should not fail if it can't parse, it simply copies the value over in that case, and leaves it as is.

**Comment by mjy** [comment]
@tmcelrath eventDate and month/day/year should be mutually exclusive.  We will likely require one or the other, then you will not have this problem?

**Comment by mjy** [comment]
@tmcelrath I'm tempted to close this and chalk it up to SOP.  https://dwc.tdwg.org/list/#dwc_eventDate says best practice is ISO, and that's what we're enforcing in eventData.

**Comment by LocoDelAssembly** [comment]
> "month" and "day" cannot have leading zeros
> ![image](https://user-images.githubusercontent.com/11411328/93932193-6a56b880-fce5-11ea-9e6d-f8dbabffb28b.png)

This is a bug. `Integer('09')` doesn't work. Have to parse differently.

DwC are not required to have both eventDate and year,month,day, but if you provide both, then the importer will check all parts agree on the same date.

**Comment by tmcelrath** [comment]
Didn't this get fixed/clarified?

**Comment by LocoDelAssembly** [comment]
It is. Before closing I need confirmation we won't use relaxed pseudo-iso datetime to allow single-digit values (and perhaps be lenient with '0' to be translated as blank instead of failing).

What would be best here?

**Comment by tmcelrath** [comment]
I'm fine with it as long as there is an error message clarifying, which I believe there is.

**Comment by tmcelrath** [comment]
0 is often exported when blank from Symbiota datasets, so I agree 'Zero' in ISO format = blank.

**Comment by tmcelrath** [comment]
Alternatively if 'Zero' in "day" part of ISO, we could import that as 1-30, or 1-31, depending on the month.

**Comment by mjy** [comment]
0 in a date should be an error I think?

**Comment by mjy** [comment]
Remember- just because Symbiota does something doesn't mean it's what we're handling.  We're handling a standard here, not everyone's edge cases.  What if somebody else has 0s in their own data by accident?

We should *absolutely* not infer translations of non legal dates to some date range.  That needs to be fixed and asserted prior to entering "tested" data I think.

**Comment by tmcelrath** [comment]
Fine with me, just be aware that some DWCA exports export blanks as zeros. As in this Symbiota "DWC Export":
![image](https://user-images.githubusercontent.com/11411328/97445314-1f8b1a80-18fb-11eb-9245-33462ee42778.png)

**Comment by tmcelrath** [comment]
@mjy agreed - just want to make sure we are all aware of what the larger community is currently doing and might cause some frustration.

**Comment by mjy** [comment]
This is a classic issue and everybody does it different.  Species Files for example stored a blank string in every field.  We do not include 0s or blank strings in the database when no value is provided because we don't knwo if the user wanted to really put 0, or the system is doing it for them.

@tmcelrath Got it.  @debpaul sound like we might need an "idiosyncracies" of exports/gotchas resource for the general communtity.... at some point down the road.

**Comment by LocoDelAssembly** [comment]
Note that the actual definition is:
> The date-time or interval during which an Event occurred. For occurrences, this is the date-time when the event was recorded. Not suitable for a time in a geological context.

Then as a **note** it says:
> Recommended best practice is to use a date that conforms to ISO 8601-1:2019.

Also provides some examples, which perhaps the ones with time spans are not really valid ISO 8601-1:2019 input. But essentially the standard is doing a (good) recommendation, but not enforcing the use of ISO dates.

**Comment by LocoDelAssembly** [comment]
I think this will be solved by either #1974 or #1861. Please re-open if you think it would not.
